insights/parsers/audit_log.py

```diff
@@ -0,0 +1,137 @@
+        >>> log.get('type=AVC')
+        'type=AVC msg=audit(1506487181.009:32794): avc:  denied  { create } for  pid=27960 comm="mongod" scontext=system_u:system_r:mongod_t:s0 tcontext=system_u:system_r:mongod_t:s0 tclass=unix_dgram_socket'
+        >>> avc_log_info = list(log.parse_lines(log.get('type=AVC')))[0]
+        >>> avc_log_info
+        {
+            'timestamp': '1506487181.009',
+            'unparsed': 'avc:  denied  { create } for',
+            'msg_ID': '32794',
+            'pid': '27960',
+            'raw_message': 'type=AVC msg=audit(1506487181.009:32794): avc:  denied  { create } for  pid=27960 comm="mongod" scontext=system_u:system_r:mongod_t:s0 tcontext=system_u:system_r:mongod_t:s0 tclass=unix_dgram_socket',
+            'comm': 'mongod',
+            'scontext': 'system_u:system_r:mongod_t:s0',
+            'tclass': 'unix_dgram_socket',
+            'type': 'AVC',
+            'tcontext': 'system_u:system_r:mongod_t:s0'
+        }
+        >>> assert avc_log_info == list(log.get_after(timestamp=date.fromtimestamp(1506047401.407)))[0]
+    """
+
+    def parse_lines(self, lines):
+        """
+        Parse log lines, e.g. from keep_scan or get, into info(dict).
+
+        Parsing logic for each line:
+
+            * First, use `shlex.split` to split by empty.
+            * Next, assert the first two key-value pair is 'type' and 'msg'.
+            * Next, parse the remained string reversly to get key-value pair data as more as possible.
+            * The left unparsed string will be stored at "unparsed".
+            * At last, wrap all key-value pairs into a dict called info.
+
+        Parameters:
+            lines(list): List of lines to parse
+
+        Yields:
+            Dictionaries corresponding to the data in each line split into
+            functional elements:
+
+            * **raw_message** - the original unparsed line
+            * **type** - the log type
+            * **timestamp** - the timestamp of the log line (as a string)
+            * **msg_ID** - the serial number behind the timestamp
+            * **unparsed** - the unparsed part in log
+
+            If the line is not regular, for some reason, then as many fields as
+            possible are pulled from the line.
+        """
+        for line in lines:
+            info = {}
+            linesp = shlex.split(line)
+
+            if len(linesp) < 2:
+                info = None
+                continue
+
+            if not (linesp[0] and linesp[0].startswith('type=')):
+                info = None
+                continue
+            info['type'] = linesp[0].lstrip('type=')
+
+            if not (linesp[1] and linesp[1].startswith('msg=audit(')):
+                info = None
+                continue
+            timestamp_id = linesp[1].lstrip('msg=audit(').rstrip('):')
+            timestamp_id_sp = timestamp_id.split(':')
+            if len(timestamp_id_sp) != 2:
+                info = None
+                continue
```

Comment:
I think you might be able to simplify this  **if / continue** logic since you've already checked if there are two items in the list and you haven't set `info` to any value?

Reply:
Yep, I can remove several `info = None`  lines.

(If I understand u correctly)   
Checking of    
`if len(linesp) < 2:`   
and    
`if len(timestamp_id_sp) != 2`   
are for different parts.